contrib/fsl_integration/sct_plugin.py

```diff
@@ -216,44 +285,530 @@ def __init__(self, parent):
+    <b>Specific citation</b>:<br>
+    <b>Usage</b>:<br>
+    You need to have the raw imaging (T1w, T2w or T2s) selected, i.e, purple line over the file name
+    and to display the image, click on the eye icon next to it. 
+    For more information, please refer to the article below.<br><br>
+    Martin et al.
+    <i>Can Microstructural MRI detect subclinical tissue injury in subjects with asymptomatic
+    cervical spinal cord compression? A prospective cohort study
+    (2018)</i>. BMJ Open. 13;8(4)e01980
+    """
+
+
+    def __init__(self, parent):
+        super(TabPanelCSA, self).__init__(parent=parent,
+                                          id_=wx.ID_ANY)
+        button_gm = wx.Button(self, id=wx.ID_ANY, label="Cross-Sectional Area")
+        button_gm.Bind(wx.EVT_BUTTON, self.onButtonCSA, id = button_gm.GetId())
+
+        lbl_contrasts = ['GM', 'WM', 'Total']
+        self.rbox_contrast = wx.RadioBox(self, label='Select tissue:',
+                                         choices=lbl_contrasts,
+                                         majorDimension=1,
+                                         style=wx.RA_SPECIFY_ROWS)
+
+        lbl_perlevel = ['No', 'Yes']
+        self.rbox_perlevel = wx.RadioBox(self, label='Perlevel:',
+                                         choices=lbl_perlevel,
+                                         majorDimension=1,
+                                         style=wx.RA_SPECIFY_ROWS)
+
+        hbox1 = wx.BoxSizer(wx.HORIZONTAL)
+        l1 = wx.StaticText(self, wx.ID_ANY, "Vertebral Levels")
+        hbox1.Add(l1, 1, wx.EXPAND | wx.ALIGN_LEFT | wx.ALL, 5)
+        self.t1 = wx.TextCtrl(self)
+        self.t1.Bind(wx.EVT_TEXT, self.OnKeyTyped)
+        hbox1.Add(self.t1, 1, wx.EXPAND | wx.ALIGN_LEFT | wx.ALL, 5)
+
+        sizer = wx.BoxSizer(wx.VERTICAL)
+        sizer.Add(self.rbox_contrast, 0, wx.ALL, 5)
+        sizer.Add(self.rbox_perlevel, 0, wx.ALL, 5)
+        sizer.Add(hbox1)
+        sizer.Add(button_gm, 0, wx.ALL, 5)
+        self.sizer_h.Add(sizer)
+        self.SetSizerAndFit(self.sizer_h)
+
+
+    def OnKeyTyped(self, event):
+        vert = event.GetString()
+
+
+    def onButtonCSA(self, event):
+        selected_overlay = displayCtx.getSelectedOverlay()
+        filename_path = selected_overlay.dataSource
+        contrast = self.rbox_contrast.GetStringSelection()
+        perlevel = self.rbox_perlevel.GetStringSelection()
+        vert = self.t1.GetValue()
+
+        print('Contrast:', contrast)
+
+        print('Verts:', vert)
+
+        print('Perlevel:',perlevel)
+
+        path_name = os.path.dirname(filename_path)
+        base_name = os.path.basename(filename_path)
+
+        fname, fext = base_name.split(os.extsep, 1)
+        seg_name = "{}_seg.{}".format(fname, fext)
+        segname_path = os.path.join(path_name, seg_name)
+
+        print('Tissue:', contrast)
+
+        if perlevel == 'No':
+
+            if contrast == 'GM':
```

Comment:
i would instead display a box where people can change the default output file name